### PR TITLE
Use only one SDK init

### PR DESCRIPTION
**Pull request by martinmitrevski**

### CODE REVIEW

File: DemoApp/StreamVideoSwiftUIApp.swift [hunk 1]
```diff
@@ -99,9 +99,11 @@ struct StreamVideoSwiftUIApp: App {
             }
             return currentUser
         }()
-        let streamVideo = try await StreamVideo(
+        let streamVideo = StreamVideo(
```
Comment (ipavlidakis) [inline DemoApp/StreamVideoSwiftUIApp.swift:102]: What's the suggested wait to retrieve information for the connection status?

File: Sources/StreamVideo/StreamVideo.swift [hunk 2]
```diff
@@ -217,18 +124,26 @@ public class StreamVideo {
             defaultAPI.middlewares.append(anonymousAuth)
         }
         self.prefetchLocation()
+        connectTask = Task {
+            if user.type == .guest {
+                do {
+                    let guestInfo = try await loadGuestUserInfo(for: user, apiKey: apiKey)
+                    self.user = guestInfo.0
```
Comment (ipavlidakis) [inline Sources/StreamVideo/StreamVideo.swift:131]: nit: Not in love with the parameter index for access

File: Sources/StreamVideo/StreamVideo.swift [hunk 3]
```diff
@@ -448,6 +353,30 @@ public class StreamVideo {
...
+        // Update the user and token provider.
+        let updatedUser = guestUserResponse.user.toUser
+        let tokenProvider = { [unowned self] result in
+            Self.loadGuestToken(
+                userId: user.id,
+                apiKey: apiKey,
+                environment: self.environment,
+                result: result
+            )
+        }
```
Comment (ipavlidakis) [inline Sources/StreamVideo/StreamVideo.swift:376]:
```suggestion
        let updatedUser = guestUserResponse.user.toUser
        let tokenProvider = { [environment = self.environment] result in
            Self.loadGuestToken(
                userId: user.id,
                apiKey: apiKey,
                environment: environment,
                result: result
            )
        }
```

File: Sources/StreamVideo/StreamVideo.swift (hunk 2 above)
Comment (martinmitrevski) [reply Sources/StreamVideo/StreamVideo.swift]: fixed

**Review by ipavlidakis — APPROVED** [review]

File: Sources/StreamVideo/StreamVideo.swift [hunk 4]
```diff
@@ -11,14 +11,16 @@ public typealias UserTokenUpdater = (UserToken) -> Void
 
 /// Main class for interacting with the `StreamVideo` SDK.
 /// Needs to be initalized with a valid api key, user and token (and token provider).
-public class StreamVideo {
+public class StreamVideo: ObservableObject {
     
-    public let user: User
+    @Published var connectionStatus: ConnectionStatus = .initialized
```
Comment (ipavlidakis) [inline Sources/StreamVideo/StreamVideo.swift:16]: We need to make this one public

**Pull request merged**